api/rest/restapi_test.go

```diff
@@ -584,6 +584,67 @@ func TestAPIPinEndpoint(t *testing.T) {
 	testBothEndpoints(t, tf)
 }
 
+type pathCase struct {
+	path    string
+	wantErr bool
+	code    int
+}
+
+var pathTestCases = []pathCase{
+	{
+		"/ipfs/QmaNJ5acV31sx8jq626qTpAWW4DXKw34aGhx53dECLvXbY?replication-min=6&replication-max=7&name=hello+there",
```

Comment:
Should we `TestIPFSPath1 + "?query" ? 

Also, worth having `PinOptions` as a separate field in the struct and constructing the query in the test with ToQuery().

Reply:
I like it the way it is. It is a nice hardcoded thing. In current form you get to see exactly what you are testing. 
Having testcases clearly visible right above the test, seems like a good pattern to me.
Also, In making it a variable, it would be somewhere else (test package).

Reply:
Then remove all the `TestIPFSPath1` from the `test` package and hardcode the paths here.

The reason I'm asking you to use `ToQuery()` is because changing the name of a parameter or adding new ones in ToQuery() will automatically update the test queries and not require the next person to figure out that some query strings were hardcoded despite having a generation function for them. The PinOptions object would still be hardcoded here in the testcase.